includes/Stories_Renderer/Renderer.php

```diff
@@ -0,0 +1,368 @@
+	 */
+	protected function render_story_with_poster( array $story_data ) {
+		$height       = ! empty( $story_data['height'] ) ? absint( $story_data['height'] ) : 600;
+		$width        = ! empty( $story_data['width'] ) ? absint( $story_data['width'] ) : 360;
+		$poster_style = sprintf( 'background-image: url(%1$s);', esc_url_raw( $story_data['poster'] ) );
+		$poster_style = ( true === $this->is_view_type( 'carousel' ) ) ?
+			sprintf( '%1$s width: %2$spx; height: %3$spx', $poster_style, $width, $height ) : $poster_style;
+
+		?>
+		<a class="<?php echo( esc_attr( "image-align-{$this->attributes['list_view_image_alignment']}" ) ); ?>"
+			href="<?php echo( esc_url_raw( $story_data['url'] ) ); ?>"
+		>
+			<div
+				class="web-stories__story-placeholder"
+				style="<?php echo esc_attr( $poster_style ); ?>"
+			></div>
+			<?php $this->get_content_overlay( $story_data ); ?>
+		</a>
+		<?php
+	}
+
+	/**
+	 * Renders a story with amp-story-player.
+	 *
+	 * @param array $story_data Story attributes. Contains information like url, height, width, etc of the story.
+	 *
+	 * @return void
+	 */
+	protected function render_story_with_story_player( array $story_data ) {
+		$height                  = ! empty( $story_data['height'] ) ? absint( $story_data['height'] ) : 600;
+		$width                   = ! empty( $story_data['width'] ) ? absint( $story_data['width'] ) : 360;
+		$player_style            = sprintf( 'width: %1$spx;height: %2$spx', $width, $height );
+		$story_player_attributes = $this->is_amp_request() ? sprintf( 'height=%d width=%d', $height, $width ) : '';
+		$poster_style            = ! empty( $story_data['poster'] ) ? sprintf( '--story-player-poster: url(%s)', $story_data['poster'] ) : '';
+		?>
+		<amp-story-player <?php echo( esc_attr( $story_player_attributes ) ); ?> style="<?php echo esc_attr( $player_style ); ?>">
+			<a href="<?php echo esc_url( $story_data['url'] ); ?>" style="<?php echo esc_attr( $poster_style ); ?>"><?php echo esc_html( $story_data['title'] ); ?></a>
+		</amp-story-player>
+
+		<?php
+		$this->get_content_overlay( $story_data );
+	}
+
+	/**
+	 * Renders the content overlay markup.
+	 *
+	 * @param array $story_data Story item data. Contains information like url, height, width, etc of the story.
+	 *
+	 * @return void
+	 */
+	protected function get_content_overlay( array $story_data ) {
+		if ( empty( $story_data['show_content_overlay'] ) || ( true !== $story_data['show_content_overlay'] ) ) {
+			return;
+		}
+
+		?>
+		<div
+			class="story-content-overlay web-stories__story-content-overlay"
+		>
+			<?php if ( ! empty( $story_data['title'] ) ) : ?>
+			<div class="story-content-overlay__title">
+				<?php
+				echo( esc_html( $story_data['title'] ) );
+				?>
+			</div>
+			<?php endif; ?>
+			<div class="story-content-overlay__author-date">
+				<?php if ( ! empty( $story_data['author'] ) ) : ?>
+				<div>
+					<?php
+					esc_html_e( 'By', 'web-stories' );
+					echo( esc_html( ' ' . $story_data['author'] ) );
```

Comment:
String concatenation is a bad practice in internationalization.

Better:

```php
/* translators: %s: author name. */
echo esc_html( sprintf( __( 'By %s', 'web-stories' ), $story_data['author'] ) );
```

same for the publish date below